commit 539503b38d9d77d654d2dd573f6f1310f3b309ee
Author: Philipp von Weitershausen <philikon@silklabs.com>
Date:   Thu Sep 8 15:06:24 2016 -0700

    react-native fix: dedupe symlinks if they point to already covered paths
    
    See https://github.com/facebook/react-native/pull/9819

diff --git a/node_modules/react-native/local-cli/server/findSymlinksPaths.js b/node_modules/react-native/local-cli/server/findSymlinksPaths.js
index 804e1d4..af64f2d 100644
--- a/node_modules/react-native/local-cli/server/findSymlinksPaths.js
+++ b/node_modules/react-native/local-cli/server/findSymlinksPaths.js
@@ -1,12 +1,25 @@
 const path = require('path');
 const fs = require('fs');
 
-module.exports = function findSymlinksPaths(lookupFolder) {
+function isSubPathOfPath(parentPath, subPath) {
+  return !path.relative(parentPath, subPath).startsWith('..' + path.sep);
+}
+
+function isSubPathOfPaths(parentPaths, subPath) {
+  return parentPaths.some(parentPath => isSubPathOfPath(parentPath, subPath));
+}
+
+/**
+ * Find and resolve symlinks in `lookupFolder`, filtering out any
+ * paths that are subpaths of `existingSearchPaths`.
+ */
+module.exports = function findSymlinksPaths(lookupFolder, existingSearchPaths) {
   const timeStart = Date.now();
   const folders = fs.readdirSync(lookupFolder);
   const resolvedSymlinks = folders.map(folder => path.resolve(lookupFolder, folder))
     .filter(folderPath => fs.lstatSync(folderPath).isSymbolicLink())
-    .map(symlink => path.resolve(process.cwd(), fs.readlinkSync(symlink)));
+    .map(symlink => path.resolve(process.cwd(), fs.readlinkSync(symlink)))
+    .filter(symlinkPath => !isSubPathOfPaths(existingSearchPaths, symlinkPath));
   const timeEnd = Date.now();
 
   console.log(`Scanning ${folders.length} folders for symlinks in ${lookupFolder} (${timeEnd - timeStart}ms)`);
diff --git a/node_modules/react-native/local-cli/server/server.js b/node_modules/react-native/local-cli/server/server.js
index 6ed2191..d244aaa 100644
--- a/node_modules/react-native/local-cli/server/server.js
+++ b/node_modules/react-native/local-cli/server/server.js
@@ -21,7 +21,7 @@ const NODE_MODULES = path.resolve(__dirname, '..', '..', '..');
 function server(argv, config, args) {
   args.projectRoots = args.projectRoots.concat(
     args.root,
-    findSymlinksPaths(NODE_MODULES)
+    findSymlinksPaths(NODE_MODULES, args.projectRoots)
   );
 
   console.log(formatBanner(
